### PR TITLE
Fix: rotate resolution decreasing on click

### PR DESCRIPTION
**Pull request by salimkanoun**
### Context
As discussed yesterday if Office hours, the click with the TrackballRotateTool was causing a permanent and additive resolution decrease

### Changes & Results

Reworked listener to do resolution decrease at drag callback.
However, this mean resolution is changed everytime a drag event occurs, it might have performance impact, as resolution decrease / reincrease could have computation cost.

### Testing

Go to Ortientation Maker Tool, click on the 3D viewport and drag, resolution doesn't decrease anymore

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

**Comment by salimkanoun** [comment]
@sedghi if the user drag and drop a new series in an existing HP, is this rotation tool going to be reinstanciated ?

If not it will mean that the mapper is going to be set to the previous series value and not the new one....

**Comment by salimkanoun** [comment]
And by the way I'm not sure about Performances,and qualify decrease / increase will be done at each drag event that are so many.

To handle more properly we will need to manage image quality reduction in drag start and drag end event

File: packages/tools/src/tools/TrackballRotateTool.ts [hunk 1]
```diff
@@ -172,6 +170,7 @@ class TrackballRotateTool extends BaseTool {
     const { rotateIncrementDegrees } = this.configuration;
     const enabledElement = getEnabledElement(element);
     const { viewport } = enabledElement;
+    this._reduceSampleDistance(viewport);
```
Comment (sedghi) [inline packages/tools/src/tools/TrackballRotateTool.ts:173]: I'm not entirely certain about the reason for invoking the `reduceSampleDistance` function during each drag operation. This function executes multiple times, potentially tens of times, within a single interaction.

Ideally, it should start before the mouse event (preMouseDown) and reset after the mouseUp event.

**Comment by salimkanoun** [comment]
@sedghi I tried but i'm unable to get the onMouseUp event, is passes throught anoter custom listener that is obscure to me.

I also tried to expose the dragstart / dragend event but didn't succeed as all mouse event seems to pass to a specific listener in cornerstone 

I think it would be better to use the native HTML even drag start / drag end, no chance to get these events in cornerstone ?

**Comment by sedghi** [comment]
@IbrahimCSAE please see how to fix this properly

**Comment by sedghi** [comment]
@salimkanoun Can you please provide reproducible steps so I can observe the incorrect behavior in the main branch?

**Comment by salimkanoun** [comment]
https://github.com/cornerstonejs/cornerstone3D/assets/25229822/3079b9a3-4e7c-40ab-92bd-c53e6fe5be34

**Comment by sedghi** [comment]
I wasn't able to push changes to this PR, so i created another PR, just FYI https://github.com/cornerstonejs/cornerstone3D/pull/1306

**Pull request closed without merging**